test/suites/storage.sh

```diff
@@ -7,10 +7,17 @@ test_storage() {
   LXD_STORAGE_DIR=$(mktemp -d -p "${TEST_DIR}" XXXXXXXXX)
   spawn_lxd "${LXD_STORAGE_DIR}" false
 
-  # edit storage and pool description
   local storage_pool storage_volume
   storage_pool="lxdtest-$(basename "${LXD_DIR}")-pool"
   storage_volume="${storage_pool}-vol"
+
+  # Check for pool name validation
+  ! lxc query -X POST --wait /1.0/storage-pools --data '{\"driver\":\"'"${lxd_backend}"'\"}' || false
+  ! lxc query -X POST --wait /1.0/storage-pools --data '{\"name\":\"-'"${storage_pool}"'\", \"driver\":\"'"${lxd_backend}"'\"}' || false
+  ! lxc storage create "${storage_pool}/" "$lxd_backend" || false
+  ! lxc storage create "${storage_pool} " "$lxd_backend" || false
```

Comment:
Nit pick: this new occurrence could use `{}`: `${lxd_backend}`
I know we are absolutely not consistent about this rule anywhere but since you've done it for the other variables :)